web/src/test/java/org/springframework/security/web/bind/support/CurrentSecurityContextArgumentResolverTests.java

```diff
@@ -0,0 +1,287 @@
+		setAuthenticationPrincipal(principal);
+		assertThat(resolver.resolveArgument(showSecurityContextErrorOnInvalidTypeImplicit(), null, null, null))
+				.isNull();
+	}
+
+	@Test
+	public void resolveArgumentSecurityContextErrorOnInvalidTypeFalse() throws Exception {
+		String principal = "invalid_type_false";
+		setAuthenticationPrincipal(principal);
+		assertThat(resolver.resolveArgument(showSecurityContextErrorOnInvalidTypeFalse(), null, null, null))
+				.isNull();
+	}
+
+	@Test
+	public void resolveArgumentSecurityContextErrorOnInvalidTypeTrue() throws Exception {
+		String principal = "invalid_type_true";
+		setAuthenticationPrincipal(principal);
+		assertThatExceptionOfType(ClassCastException.class).isThrownBy(() -> resolver.resolveArgument(showSecurityContextErrorOnInvalidTypeTrue(), null,
+				null, null));
+	}
+
+	private MethodParameter showSecurityContextNoAnnotation() {
+		return getMethodParameter("showSecurityContextNoAnnotation", String.class);
+	}
+
+	private MethodParameter showSecurityContextAnnotation() {
+		return getMethodParameter("showSecurityContextAnnotation", SecurityContext.class);
+	}
+
+	private MethodParameter showAnnotationWithCustomSecurityContext() {
+		return getMethodParameter("showAnnotationWithCustomSecurityContext", CustomSecurityContext.class);
+	}
+
+	private MethodParameter showAnnotationWithCustomSecurityContextTypeMatch() {
+		return getMethodParameter("showAnnotationWithCustomSecurityContextTypeMatch", SecurityContext.class);
+	}
+
+	private MethodParameter showSecurityContextAuthenticationAnnotation() {
+		return getMethodParameter("showSecurityContextAuthenticationAnnotation", Authentication.class);
+	}
+
+	private MethodParameter showSecurityContextAuthenticationWithOptionalPrincipal() {
+		return getMethodParameter("showSecurityContextAuthenticationWithOptionalPrincipal", Object.class);
+	}
+
+	private MethodParameter showSecurityContextAuthenticationWithPrincipal() {
+		return getMethodParameter("showSecurityContextAuthenticationWithPrincipal", Object.class);
+	}
+
+	private MethodParameter showSecurityContextWithUserDetail() {
+		return getMethodParameter("showSecurityContextWithUserDetail", Object.class);
+	}
+
+	private MethodParameter showSecurityContextErrorOnInvalidTypeImplicit() {
+		return getMethodParameter("showSecurityContextErrorOnInvalidTypeImplicit", String.class);
+	}
+
+	private MethodParameter showSecurityContextErrorOnInvalidTypeFalse() {
+		return getMethodParameter("showSecurityContextErrorOnInvalidTypeFalse", String.class);
+	}
+
+	private MethodParameter showSecurityContextErrorOnInvalidTypeTrue() {
+		return getMethodParameter("showSecurityContextErrorOnInvalidTypeTrue", String.class);
+	}
+
+	private MethodParameter getMethodParameter(String methodName, Class<?>... paramTypes) {
+		Method method = ReflectionUtils.findMethod(TestController.class, methodName,
+				paramTypes);
+		return new MethodParameter(method, 0);
+	}
+
+	public static class TestController {
```

Comment:
I think a test for meta-annotations is missing. Since this is a more complex class, it's better to make sure we are well covered.

Reply:
can you give me some examples? or `The Existing Annotation for reference`,  thanks

Reply:
For example, if you do:

```java
@Retention(RUNTIME)
@CurrentSecurityContext(expression="authentication")
public @interface CurrentAuthentication
```

And then

```java
public String myMethod(@CurrentAuthentication Authentication auth) {
}
```

Then, the resolution should still work.

Reply:
fixed